cmd/deploy.go

```diff
@@ -229,14 +225,8 @@ func deployFn(_ *cobra.Command, _ []string) error {
 		nodesWg.Wait()
 	}
 
-	log.Debug("containers created, retrieving state and IP addresses...")
-	// updating nodes with runtime information such as IP addresses assigned by the runtime dynamically
-	for _, n := range c.Nodes {
-		err = n.UpdateConfigWithRuntimeInfo(ctx)
-		if err != nil {
-			log.Errorf("failed to update node runtime information for node %s: %v", n.Config().ShortName, err)
-		}
-	}
+	// write to log
+	execCollection.Log()
```

Comment:
is there a need to return execCollection to the caller (deploy)
maybe the log should happen where the exec collection is populated/executed?

Reply:
I wanted to make sure, that the exec output is printed as a collection of outputs en bloc.
So thats why we use the collection here.
As you see, we have the `WaitGroup` just right below, because the main program flow is simply spawning go funcs to create the containers and uses the WaitGroup below, to "sync" with the async container creations. 
So only when that WaitGroup unblocks we can be sure all the output is generated and print it as a consecutive block rather then having it scattered all over the log.

https://github.com/srl-labs/containerlab/blob/6c009855625a3afe9ef05da9835879d6c6d40e16/cmd/deploy.go#L219-L229